Displays memory usage statistics after build

mvgrimes opened the pull request:
Given the limited memory of most Arduino boards, knowing a sketch's
memory usage is important to developers. The Arduino IDE displays this
and provides warnings when memory usage is high. The IDE includes a
tool (avr-size) that calculates memory usage.

This patch runs the tool and compares memory (both text and data) to the
maximums as stored in the boards.txt file.

mvgrimes committed:
Displays memory usage statistics after build

Given the limited memory of most Arduino boards, knowing a sketch's
memory usage is important to developers. The Arduino IDE displays this
and provides warnings when memory usage is high. The IDE includes a
tool (avr-size) that calculates memory usage.

This patch runs the tool and compares memory (both text and data) to the
maximums as stored in the boards.txt file.

scottdarch commented on ano/commands/build.py:
```diff
@@ -259,6 +266,23 @@ def _scan_dependencies(self, dirName, lib_dirs, inc_flags):
+        boardVariant = args.cpu if ('cpu' in args) else None;
+        maximum_size = int(BoardModels.getValueForVariant(board, boardVariant,
+            'upload', 'maximum_size'))
+        maximum_data = int(BoardModels.getValueForVariant(board, boardVariant,
```
maximum_data_size appears to be either newer or optional. Can you handle the KeyError thrown from this line for older IDE versions?

mvgrimes committed:
Improves error handling, memory size calcs and warnings messages

- maximum_data_size is either newish or optional, so we catch any
  KeyErrors raised when retrieving values from boards.txt.
- FLASH memory usages is actually text + data.
- SRAM memory usage is actually is data + bss.
- The memory usage statements now mirror the Arduino IDE's message syntax.
- Warnings are triggered when the sketch uses 100% of FLASH or 75% of SRAM
  (consistent with the Arduino IDE).

photodude commented:
:+1:
